keras_spiking/constraints.py

```diff
@@ -0,0 +1,36 @@
+"""
+Custom constraints for weight tensors in Keras models.
+"""
+
+import tensorflow as tf
+
+
+class Mean(tf.keras.constraints.Constraint):
+    """Constrains weight tensors to be their mean.
+
+    Parameters
+    ----------
+    axis : int
+        Axis used to compute the mean and repeat its value. Defaults to the last axis.
+    non_neg : boolean
+        Whether to constrain the resulting mean to be non-negative as well.
+    """
+
+    def __init__(self, axis=-1, non_neg=False):
+        self.axis = axis
+        self.non_neg = non_neg
+        self._post_constraint = tf.keras.constraints.get("non_neg" if non_neg else None)
+
+    def __call__(self, w):
+        r = tf.repeat(
+            input=tf.math.reduce_mean(w, axis=self.axis, keepdims=True),
+            repeats=tf.shape(w)[self.axis],
+            axis=self.axis,
+        )
+        if self._post_constraint is not None:
+            r = self._post_constraint(r)
+        return r
+
+    def get_config(self):
+        """Return config of layer (for serialization during model saving/loading)."""
+        return {"axis": self.axis, "non_neg": self.non_neg}
```

Comment:
This line is uncovered. Should probably add a `test_constraints.py` file that tests this, as well as a basic test of the constraint on e.g. a dense layer (which can be pretty similar to the one you added, it'd just be nice to have a unit test that's independent of the application to `LowpassCell`).

Reply:
I believe it will be convered when that failed test is made to pass (see comment in main PR).

Reply:
Wait sorry scratch that. I would need to actually use the constraint in that test to cover this line.